core/src/test/java/io/confluent/kafka/schemaregistry/leaderelector/kafka/SchemaRegistryCoordinatorTest.java

```diff
@@ -153,148 +221,180 @@ public void testMetadata() {
+    member1Coordinator.setAssignmentSnapshot(null);
+    initiateJoinSyncFlow(MEMBER_0, member0Coordinator, groupMembership, true, MEMBER_0, MEMBER_0_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_2, member2Coordinator, groupMembership, false, MEMBER_0, MEMBER_0_INFO, Assignment.NO_ERROR);
+    validate(member0Coordinator, member0RebalanceListener, 5, 6, false, MEMBER_0, MEMBER_0_INFO);
+    validate(member2Coordinator, member2RebalanceListener, 1, 3, false, MEMBER_0, MEMBER_0_INFO);
+
+    // member 1 re-joins group
+    groupMembership.put(MEMBER_1, MEMBER_1_INFO);
+    initiateJoinSyncFlow(MEMBER_0, member0Coordinator, groupMembership, true, MEMBER_0, MEMBER_0_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_1, member1Coordinator, groupMembership, false, MEMBER_0, MEMBER_0_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_2, member2Coordinator, groupMembership, false, MEMBER_0, MEMBER_0_INFO, Assignment.NO_ERROR);
+    validate(member0Coordinator, member0RebalanceListener, 6, 7, false, MEMBER_0, MEMBER_0_INFO);
+    validate(member1Coordinator, member1RebalanceListener, 3, 5, false, MEMBER_0, MEMBER_0_INFO);
+    validate(member2Coordinator, member2RebalanceListener, 2, 4, false, MEMBER_0, MEMBER_0_INFO);
+
+    // member 0 leaves group
+    groupMembership.remove(MEMBER_0);
+    member0Coordinator.setAssignmentSnapshot(null);
+    member0Coordinator.getIdentity().setLeader(false);
+    initiateJoinSyncFlow(MEMBER_1, member1Coordinator, groupMembership, true, MEMBER_1, MEMBER_1_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_2, member2Coordinator, groupMembership, false, MEMBER_1, MEMBER_1_INFO, Assignment.NO_ERROR);
+    validate(member1Coordinator, member1RebalanceListener, 4, 6, false, MEMBER_1, MEMBER_1_INFO);
+    validate(member2Coordinator, member2RebalanceListener, 3, 5, false, MEMBER_1, MEMBER_1_INFO);
+
+    // member 0 re-joins group
+    groupMembership.put(MEMBER_0, MEMBER_0_INFO);
+    initiateJoinSyncFlow(MEMBER_0, member0Coordinator, groupMembership, true, MEMBER_0, MEMBER_1_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_1, member1Coordinator, groupMembership, false, MEMBER_0, MEMBER_1_INFO, Assignment.NO_ERROR);
+    initiateJoinSyncFlow(MEMBER_2, member2Coordinator, groupMembership, false, MEMBER_0, MEMBER_1_INFO, Assignment.NO_ERROR);
+    validate(member0Coordinator, member0RebalanceListener, 6, 8, false, MEMBER_0, MEMBER_1_INFO);
+    validate(member1Coordinator, member1RebalanceListener, 5, 7, false, MEMBER_0, MEMBER_1_INFO);
+    validate(member2Coordinator, member2RebalanceListener, 4, 6, false, MEMBER_0, MEMBER_1_INFO);
+  }
+
+  private void validate(SchemaRegistryCoordinator memberCoordinator,
+                        MockRebalanceListener memberRebalanceListener,
+                        int revokedCount,
+                        int assignedCount,
+                        boolean assignmentFailed,
+                        String groupLeader,
```

Comment:
What is difference between groupLeader and leaderIdentity?

Reply:
groupLeader is one of the SR nodes which does the leadership assignment. leaderIdentity is the node that is the leader from the SR application perspective. it's the node that is responsible for the write operations.

Reply:
Thanks. Can you explain 
1. what is significance of groupLeader?
2. why in both test cases when Member 0 rejoins the group, Member 0 is the one who does leadership assignment(groupLeader) thought Member 1 is the current leader?

Reply:
1. groupLeader is the only one that does the assignment. it's to avoid multiple members all doing leadership assignments and coming up with different leaders. the other members wait for the assignment from the groupLeader.
2. picking member-0 is arbitrary. the logic to assign the SR leader wouldn't change regardless of who the groupLeader is.

Reply:
Thanks. That help.

Just my thoughts, feel free to ignore. I feel it would be easier to rename groupLeader to leaderElector. My first thought was groupLeader is the leader in SR which does the writes.